google/cloud/dataproc_spark_connect/environment.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2024 Google LLC
+#
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# environment_utils.py
+
+# environment_utils.py
+
+import os
+import json
+import importlib
+import logging
+from typing import Callable, Tuple, List
+
+from google.cloud.dataproc_spark_connect.constants import (
+    CLIENT_LABEL_VALUE_UNKNOWN,
+    CLIENT_LABEL_VALUE_COLAB,
+    CLIENT_LABEL_VALUE_COLAB_ENTERPRISE,
+    CLIENT_LABEL_VALUE_WORKBENCH,
+    CLIENT_LABEL_VALUE_BQ_STUDIO,
+    CLIENT_LABEL_VALUE_VSCODE,
+    CLIENT_LABEL_VALUE_JUPYTER,
+)
+
+# Known extension identifier
+GOOGLE_CLOUD_CODE_EXTENSION = "googlecloudtools.cloudcode"
+
+
+def _is_vscode_extension_installed(extension_id: str) -> bool:
+    """
+    Scan the VS Code extensions
+    folder for a given ID.
+    """
+    try:
+        home = os.path.expanduser("~")
+        ext_dir = os.path.join(home, ".vscode", "extensions")
+        if not os.path.isdir(ext_dir):
+            return False
+
+        for folder in os.listdir(ext_dir):
+            if not folder.startswith(extension_id + "-"):
+                continue
+            manifest = os.path.join(ext_dir, folder, "package.json")
+            if os.path.isfile(manifest):
+                with open(manifest, encoding="utf-8") as f:
+                    json.load(f)
+                return True
+    except Exception:
+        logging.exception(
+            "Failed to detect extension %s",
+            extension_id,
+        )
```

Comment:
Do we need to log anything? I think that it may be better to fail silently to not confuse users?